core/domain/state_domain.py

```diff
@@ -173,6 +174,187 @@ def validate(self):
         self.hint_content.validate()
 
 
+class Image(object):
+    """Value object representing the image for the state."""
+    def __init__(self, src, placeholder, author_id, instructions):
+        """Constructs an Image domain object.
+
+        Args:
+            src: str. The source of an image.
+            placeholder: bool. Is image a placeholder.
+            author_id: str. Id of an author which uploads an image.
+            instructions: str. Instructions for the image which needs in place
+                of a placeholder.
+        """
+        self.src = src
+        self.placeholder = placeholder
+        self.author_id = author_id
+        self.instructions = instructions
+        self.validate()
+
+    def to_dict(self):
+        """Returns a dict representing this Image domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of Image instance.
+        """
+        image = {}
+        image['src'] = self.src
+        image['placeholder'] = self.placeholder
+        image['author_id'] = self.author_id
+        image['instructions'] = self.instructions
+        return image
+
+    @classmethod
+    def from_dict(cls, image_info):
+        """Return an Imagedomain object from a dict.
+
+        Args:
+            image_info: dict. The dict representation of
+                Image object.
+
+        Returns:
+            Image. The corresponding ImageAssets domain
+            object.
+        """
+        src = image_info['src']
+        placeholder = image_info['placeholder']
+        author_id = image_info['author_id']
+        instructions = image_info['instructions']
+
+        return cls(src, placeholder, author_id, instructions)
+
+    def validate(self):
+        """Validates all properties of an Image.
+
+        Raises:
+            ValidationError: One or more attributes of the ImageAssets are not
+            valid.
+        """
+        # src_re = r'^[A-Za-z0-9+/]*\.((png)|(jpeg)|(gif)|(jpg))$'
+        # if not re.match(src_re, self.src):
+        #     raise utils.ValidationError(
+        #         'Invalid image_src, received %s' % self.src)
+        if not isinstance(self.src, basestring):
+            raise utils.ValidationError(
```

Comment:
Can you please check whether we have any pattern check for image src?

Reply:
done